### PR TITLE
Fix Any json encoding/decoding in php

### PR DESCRIPTION
**Pull request by TeBoring**

### CODE REVIEW

**Comment by TeBoring** [comment]
@michaelbausor Could you help take a look?

File: php/tests/encode_decode_test.php [hunk 1]
```diff
@@ -915,4 +917,127 @@ public function testEncodeTopLevelStruct()
...
+        $packed->setSeconds(946684800);
+        $packed->setNanos(123456789);
+        $m = new Any();
+        $m->pack($packed);
+        $this->assertSame(
+            "{\"@type\":\"type.googleapis.com/google.protobuf.Timestamp\"," .
+            "\"value\":\"2000-01-01T00:00:00.123456789Z\"}",
+            $m->serializeToJsonString());
+    }
+
```
Comment (michaelbausor) [inline php/tests/encode_decode_test.php]: What do we expect to happen if we try to deserialize a json Any when the packed type in unknown? Can we test for that? Ideally we would still get an Any object, but with unknown data inside it. But maybe that is not possible?

Comment (TeBoring) [reply php/tests/encode_decode_test.php]: I think the behavior should be failure.

Comment (TeBoring) [reply php/tests/encode_decode_test.php]: Added test to verify unknown message throws exception.

**Comment by TeBoring** [comment]
I don't think the go and objc test is related to this change.

**Comment by TeBoring** [comment]
@michaelbausor, fixed your comments. Any more comments?

**Pull request merged**